dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java

```diff
@@ -0,0 +1,239 @@
+package com.example.michael.mqtttest;
+
+import android.app.Service;
+import android.content.Intent;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import org.eclipse.paho.android.service.MqttAndroidClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * @desc - This is the work horse for acceleration data broadcasting. When
+ *         this service is run,it will attempt to connect to an MQTT broker at
+ *         the IP address provided by the activity calling this service. The
+ *         AccelerationListener class in this service is responsible for
+ *         calling the 'publish' method of the MQTT client to send data to the
+ *         MQTT broker. The acceleration listener is also responsible for
+ *         detecting when the device is disconnected and perpetually
+ *         attempting to re-establish connection.
+ *
+ *         TO-DO:
+ *         We should consider moving the responsibility of publishing data
+ *         from the AccelerationListener to a running thread that triggers a
+ *         publish event in discrete time and have the AccelerationListener
+ *         change the values published rather than tirggering publications on
+ *         acceleration events based on a delay
+ */
+public class BroadcastService extends Service {
+    /* Use this to determine if the service is running */
+    public static Boolean broadcastServiceIsRunning = false;
```

Comment:
If we have time to refactor and stuff, it might be good to make this a singleton. Right now, we are using this field to ensure it's singleton when we use the class. this can be very error prone.